File: conformance/bundles/11-emoji-offsets/greet.steps.rs
//! Rust sibling of `greet.steps.ts` (bundle `11-emoji-offsets`).

use var::{Handler, Registry, Steps, Value};

pub const FILE: &str = "greet.steps.rs";

pub fn register(r: Registry) -> Registry {
    let mut s = Steps::from_registry(r);
    // The list item is followed by a table, appended as a trailing arg, so this
    // sensor's slots are {string} + the table (returns nothing → passes).
    s.sensor(
        "I greet {string}",
        FILE,
        1,
        Handler::sync2(|_state, _name, _table| Ok(None)),
    );
    s.into_registry()
}

pub fn state() -> Value {
    Value::Null
}

File: conformance/bundles/11-emoji-offsets/greet.steps.cs
// C# sibling of greet.steps.ts / .rs (bundle 11-emoji-offsets).
using Varar;
using Varar.Core;

namespace Varar.Corpus.B11;

public static class GreetSteps
{
    public static Registry Register(Registry r)
    {
        var s = Steps.From(r);

        // The list item is followed by a table, appended as a trailing arg, so this sensor's slots
        // are {string} + the table (returns nothing → passes).
        s.Sensor("I greet {string}", (state, name, table) => null);

        return s.ToRegistry();
    }

    public static Value State() => Value.Null;
}
